### PR TITLE
[ENH] in `ExpandingGreedySplitter`, allow `float` `step_size`

### PR DESCRIPTION
**Pull request by fkiraly**
This PR adds logic to allow `float` `step_size` in `ExpandingGreedySplitter`.

Also makes minor changes:

* docstring formatting
* adds missing author credit for @davidgilbertson 

Questions for reviewers:

* I originally added this to help https://github.com/sktime/sktime/issues/5107, by later making `temporal_train_test_split` default to `ExpandingGreedySplitter` instances. Thi is no longer needed for #5107 though

* is there a more elegant algorithm for the fraction case along the lines of @davidgilbertson's neat algorithm for the integer case, perhaps @davidgilbertson can see this? Also, would you be fine with adding the `float` option?

Also FYI @BensHamza (splitters) and @hazrulakmal (for benchmarking and `temporal_train_test_split` connection).

### CODE REVIEW

File: sktime/split/expandinggreedy.py [hunk 1]
```diff
@@ -16,21 +17,25 @@
...
     Parameters
     ----------
-    test_size : int
-        The number of steps included in the test set of each fold.
+    test_size : int or float
+        If int: the number of steps included in the test set of each fold.
+            Formally, steps are consecutive ``iloc`` indices.
+        If float: the proportion of steps included in the test set of each fold,
+            as a proportion of the total number of index values.
+            Cave: not the ``loc`` proportion between start and end.
```
Comment (yarnabrina) [inline sktime/split/expandinggreedy.py:38]: Sorry, I don't follow this caveat. Can you please explain?

File: sktime/split/expandinggreedy.py [hunk 2]
```diff
@@ -39,7 +44,7 @@ class ExpandingGreedySplitter(BaseSplitter):
     Examples
     --------
     >>> import numpy as np
-    >>> from sktime.split import ExpandingGreedySplitter
+    >>> from sktime.forecasting.model_selection import ExpandingGreedySplitter
```
Comment (yarnabrina) [inline sktime/split/expandinggreedy.py:47]: A bit confused - I thought we are moving to `sktime.split`? Does that import not work? I see it's exported in `sktime/split/__init__.py` module of `expandinggreedysplitter-float` branch as well.

File: sktime/split/expandinggreedy.py [hunk 3]
```diff
@@ -59,18 +64,29 @@ def __init__(self, test_size: int, folds: int = 5, step_length: int = None):
...
 
+        # no algorithm implemented that is faster for float than naive iteration
+        if isinstance(test_size, float):
+            self.set_tags(**{"split_hierarchical": False})
+
     def _split(self, y: pd.Index) -> SPLIT_GENERATOR_TYPE:
+        test_size = self.test_size
+
+        if isinstance(test_size, float):
+            _test_size = round(len(y) * test_size)
```
Comment (yarnabrina) [inline sktime/split/expandinggreedy.py:75]: I saw you're using floor in #5330, so just a suggestion: `math.floor` instead of `round`?

File: sktime/split/expandinggreedy.py (hunk 1 above)
Comment (fkiraly) [reply sktime/split/expandinggreedy.py:38]: If the indices are, say, `[1, 2, 3, 4, 100]`, then the last 40% by `loc` would be only `100` (everything at or above 60), but by `iloc` it would be `[4, 100]` (everything at or above the 4-th index)

File: sktime/split/expandinggreedy.py (hunk 2 above)
Comment (fkiraly) [reply sktime/split/expandinggreedy.py:47]: accidental revert

File: sktime/split/expandinggreedy.py (hunk 3 above)
Comment (fkiraly) [reply sktime/split/expandinggreedy.py:75]: for test set, I think it should be `ceil` for consistency, but agreed

File: sktime/split/expandinggreedy.py (hunk 2 above)
Comment (fkiraly) [reply sktime/split/expandinggreedy.py]: fixed

File: sktime/split/expandinggreedy.py (hunk 3 above)
Comment (fkiraly) [reply sktime/split/expandinggreedy.py]: fixed

**Review by yarnabrina — APPROVED** [review]

**Comment by fkiraly** [comment]
(we should wait for @davidgilbertson to comment before merging as he owns the splitter)

**Comment by yarnabrina** [comment]
rebased with latest `main` - if all checks pass, is it good to merge now?

**Comment by fkiraly** [comment]
> if all checks pass, is it good to merge now?

I suppose @davidgilbertson is not watching/following, so I'd say yes. Maybe after a few days, in case he missed the first ping.
